File: MessageUClient/PublicKeyResponse.cpp
#include "PublicKeyResponse.h"
#include <string>
#include "utils.h"

PublicKeyResponse::PublicKeyResponse(const BaseResponse& header, std::string& payload)
	: BaseResponse(header.getVersion(), header.getCode(), header.getPayloadSize())
{
	try {
		parsePayload(payload);
	}
	catch (const std::exception& e) {
		std::cout << e.what() << std::endl;
	}
}	


PublicKeyResponse::~PublicKeyResponse()
{
}

std::string PublicKeyResponse::getTargetPublicKey() 
{
	return targetPublicKey;
}

std::string PublicKeyResponse::getTargetClientId() 
{
	return targetClientId;
}


void PublicKeyResponse::parsePayload(std::string& payload)
{	
	this -> targetClientId = payload.substr(0, CLIENT_ID_SIZE);
	this -> targetPublicKey = payload.substr(CLIENT_ID_SIZE, PUBLIC_KEY_SIZE);
}

File: MessageUServer/models/public_key_response.py
from models.base_response import Base_Response
from utils.defines import *



class Public_Key_Response(Base_Response):
    def __init__(self, target_client_id, target_public_key):
        super().__init__(Response_Code.PUBLIC_KEY.value[0], CLIENT_ID_SIZE + PUBLIC_KEY_SIZE)
        self.target_client_id = target_client_id
        self.target_public_key = target_public_key


    def get_binary_response(self):
        header = self.get_binary_header()
        return header + self.target_client_id + self.target_public_key


